operational_analysis/methods/quality_check_automation.py

```diff
@@ -1,120 +1,613 @@
+
+        self._df[self._t] = pd_dt_col
+        self._df = self._df.set_index(pd.DatetimeIndex(self._df[self._t_local]))
 
+        # Create the UTC-converted time-stamp
+        try:
+            self._df[self._t_utc] = pd.to_datetime(
+                [el.astimezone(tz.tzutc()) for el in dt_col]
+            ).tz_convert("UTC")
+        except AttributeError:  # catches numpy datetime error for astimezone() not existing
+            self._df = self._df.tz_convert("UTC")
+            self._df[self._t_utc] = self._df.index
+
+        # Adjust the index name to reflect the change to a UTC-based timestamp
+        self._df.index.name = self._t_utc
+
+        self._df = self._determine_offset_dst(self._df)
+
+    @abstractmethod
+    @logged_method_call
+    def run(self):
+        """
+        Run the QC analysis functions in order by calling this function. In the base
+        class, this operates as a basic demonstration of features, so it is encouraged
+        that users devise a method to enhance their data checking.
+        """
         logger.info("Identifying Time Duplications")
         self.dup_time_identification()
         logger.info("Identifying Time Gaps")
         self.gap_time_identification()
-        logger.info("Grabbing DST Transition Times")
-        self.create_dst_df()
-
-        if self._check_tz:
-            logger.info("Evaluating timezone deviation from UTC")
-            self.ws_diurnal_prep()
-            self.corr_df_calc()
-
         logger.info("Isolating Extrema Values")
         self.max_min()
         logger.info("QC Diagnostic Complete")
 
     def dup_time_identification(self):
         """
         This function identifies any time duplications in the dataset.
+        """
+        self._time_duplications = self._df.loc[
+            self._df.duplicated(subset=[self._id, self._t]), self._t
+        ]
+
+        self._time_duplications_local = self._df.loc[
+            self._df.duplicated(subset=[self._id, self._t_local]), self._t_local
+        ]
+
+        self._time_duplications_utc = self._df.loc[
+            self._df.duplicated(subset=[self._id, self._t_utc]), self._t_utc
+        ]
+
+    def gap_time_identification(self, which="original"):
+        """
+        This function identifies any time gaps in the dataset.
 
         Args:
-        (None)
+            which (:obj:`str`): One of "original" (default), "local", or "utc" to serve
+                as the timestamp basis for the gap check. Original is used by
+                default and in case of invalid input.
 
         Returns:
         (None)
         """
-        self._time_duplications = self._df.loc[
-            self._df.duplicated(subset=[self._id, self._t]), self._t
-        ]
+        which = which.lower().strip()
+        col_map = dict(original=self._t, local=self._t_local, utc=self._t_utc)
+        which = "original" if which not in col_map else which
```

Comment:
just noticed that "which" is no longer used in this function, right?

Reply:
This was a leftover, I've just updated it to be like the duplication method, and added some commentary in the examples to highlight this functionality. Also, now that we're on the same page for the example commentary, this should hopefully be all set now!